src/api/statusMatrix.js

```diff
@@ -176,11 +178,13 @@ export async function statusMatrix({
         // Ignore ignored files, but only if they are not already tracked.
         if (!head && !stage && workdir) {
           if (
-            await GitIgnoreManager.isIgnored({
-              fs,
-              dir,
-              filepath,
-            })
+            ignored
+              ? false
+              : await GitIgnoreManager.isIgnored({
+                  fs,
+                  dir,
+                  filepath,
+                })
           ) {
```

Comment:
I would use:

```
ignored: shouldIgnore = false,
```   

```javascript
if (!head && !stage && workdir) {
   if (!shouldIgnore) {
      const isIgnored = await GitIgnoreManager.isIgnored({
        fs,
        dir,
        filepath,
      });
      if (isIgnored) {
         return null
      }
   }
}
```

Reply:
Also while you're editing this file, can you check (and maybe write a test) for a status matrix with an array of strings? This looks like it only works the first single `filepath`.